internal/store/cluster.go

```diff
@@ -0,0 +1,111 @@
+package store
+
+import (
+	"database/sql"
+	"encoding/json"
+	"time"
+
+	sq "github.com/Masterminds/squirrel"
+	"github.com/mattermost/mattermost-server/model"
+	"github.com/pkg/errors"
+)
+
+// Cluster represents a Kubernetes cluster as tracked by the store.
+type Cluster struct {
+	ID                  string
+	Provider            string
+	Provisioner         string
+	ProviderMetadata    []byte
+	ProvisionerMetadata []byte
+	AllowInstallations  bool
+	CreateAt            time.Time
+	DeleteAt            *time.Time
```

Comment:
Yeah, this is probably unnecessary. I had really wanted to use `time.Time` instead of just an integer, but I've already had to jump through hoops on this one. Will converge back to just an integer.